inspire_classifier/config.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Classifier Configuration."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+
+CLASSIFIER_MAXIMUM_VOCABULARY_SIZE = 60000
+CLASSIFIER_MINIMUM_WORD_FREQUENCY = 2
+CLASSIFIER_VALIDATION_DATA_FRACTION = 0.1
+CLASSIFIER_CLASSIFICATION_CLASSES = ['Rejected', 'NonCore', 'Core']
+
+### FIXME: Check and change the classifier base path
+CLASSIFIER_BASE_PATH = os.path.join(app.instance_path, 'inspire-classifier')
+CLASSIFIER_DATA_DIR = os.path.join(CLASSIFIER_BASE_PATH, 'data')
+CLASSIFIER_MODELS_DIR = os.path.join(CLASSIFIER_BASE_PATH, 'models')
+CLASSIFIER_LANGUAGE_MODEL_DIR = os.path.join(CLASSIFIER_MODELS_DIR, 'language_model')
+CLASSIFIER_CLASSIFIER_MODEL_DIR = os.path.join(CLASSIFIER_MODELS_DIR, 'classifier_model')
+CLASSIFIER_LANGUAGE_MODEL_DATA_DIR = os.path.join(CLASSIFIER_DATA_DIR, 'language_model_data')
+CLASSIFIER_CLASSIFIER_DATA_DIR = os.path.join(CLASSIFIER_DATA_DIR, 'classifier_data')
+
+CLASSIFIER_DATAFRAME_PATH =  os.path.join(CLASSIFIER_DATA_DIR, 'inspire_data.df')
+CLASSIFIER_PRETRAINED_LANGUAGE_MODEL_PATH = os.path.join(
+    CLASSIFIER_LANGUAGE_MODEL_DIR, 'wikitext_103', 'fwd_wt103.h5')
+CLASSIFIER_PRETRAINED_LANGUAGE_MODEL_ENCODER_PATH = os.path.join(
+    CLASSIFIER_LANGUAGE_MODEL_DIR, 'wikitext_103', 'fwd_wt103_enc.h5')
+CLASSIFIER_FINETUNED_LANGUAGE_MODEL_PATH = os.path.join(
+    CLASSIFIER_LANGUAGE_MODEL_DIR, 'finetuned_language_model.h5')
+CLASSIFIER_FINETUNED_LANGUAGE_MODEL_ENCODER_PATH = os.path.join(
+    CLASSIFIER_LANGUAGE_MODEL_DIR, 'finetuned_language_encoder_model.h5')
+CLASSIFIER_TRAINED_CLASSIFIER_PATH = os.path.join(
+    CLASSIFIER_CLASSIFIER_MODEL_DIR, 'trained_classifier_model.h5')
+CLASSIFIER_WIKITEXT103_ITOS_PATH = os.path.join(
+    CLASSIFIER_LANGUAGE_MODEL_DIR, 'wikitext_103', 'itos_wt103.pkl')
+CLASSIFIER_DATA_ITOS_PATH = os.path.join(CLASSIFIER_DATA_DIR, 'data_itos.pkl')
```

Comment:
The question is: do you need a place to store some constants referring to file paths, or do you expect that all these paths will need to be changed dynamically? If a path falls into the first case, then you don't need to make it configurable, if it's the latter you do.

Maybe you only need to have a couple of configurable dirs (data and model) here, and have the rest relative to them for naming, instead of having all the intermediate things.
So we would have:
```
from pathlib import Path
from Flask import current_app

CLASSIFIER_BASE_PATH = Path(current_app.instance_path)
CLASSIFIER_DATA_DIR = CLASSIFIER_BASE_PATH / 'data'
CLASSIFIER_MODELS_DIR = CLASSIFIER_BASE_PATH / 'models'

CLASSIFIER_DATAFRAME_PATH = CLASSIFIER_DATA_DIR / 'inspire_data.df'
CLASSIFIER_PRETRAINED_LANGUAGE_MODEL_PATH = CLASSIFIER_MODELS_DIR / 'language_model' / 'wikitext_103' / 'fwd_wt103.h5'
# and so on ...
```